Inventory Management

chloech314 opened the pull request:
1. Added item pickup logic
2. Added item classes (itemdata and collectibleitem)
3. Added inventory classes for mouse and mecha
4. Added UI to display inventory for both characters

All changes were made in a duplicate, new scene: Inventory Test Scene

chrstinalin commented on Assets/Scripts/Player/PlayerItemPickup/MechaPickup.cs:
```diff
@@ -0,0 +1,18 @@
+using UnityEngine;
```
I'd put the contents of this in MechInventoryManager

chrstinalin commented on Assets/Scripts/Player/PlayerItemPickup/MousePickup.cs:
```diff
@@ -0,0 +1,40 @@
+using UnityEngine;
```
Same here, anything preventing this from being condensed into MouseInventoryManager?

chrstinalin commented on Assets/Scripts/UI/MechaInventoryText.cs:
```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
```
The UI .cs files also look like they could be condensed into a single UIManager.

chloech314 commented:
Changes made 9/16/2025:

- Pickup and inventory scripts are now condensed for the respective characters
- UI Manager created to manage the inventory sprite and text
- TMP package deleted: legacy text is now used instead, which requires no install
- Covered edge case so that the item in the mouse's inventory is now added to the mecha's inventory when the mouse enters the mecha
- Added RemoveItem to the abstract class of inventory manager; the mouse deletes the item in inventory when this is called. This function was not implemented for mecha, assuming they don't need to drop anything (for now). 
- Added R key / joystick button 2 mapping -> DropItem (Again, only mouse can do this for now)
- Inventory manager now uses a list + maxnumitems variable
